Added HttpRequestor to make http request to API services, added full schedule processing

alexzhyshko opened the pull request:
Task: Familiarize with KPI schedule API and implement appropriative interface

StudentLgotnik commented on src/main/java/student/enterprise/project/dto/ChangeDTO.java:
```diff
@@ -0,0 +1,129 @@
+    private String dayName = "";
+    
+    //
+    public void setDayName(String name) {
```
Геттеры и сеттеры не нужны, у тебя уже есть аннотация Data

alexzhyshko replied:
Тут нужны, это сеттеры для внутренних классов для удобного изменения параметров классов. Сеттеры дл яэтого класса уберу.

StudentLgotnik replied:
Твой класс нарушает Single responsibility принцип, этот класс служитдля хранения и изменения состояния ченжа(предмета), а ты еще возложил на него функции редактирования другого класса.
Например, ты сначала должен определить состояние LessonDTO(через его геттеры и сеттеры) а потом уже положить его сюда через метод setLessonDTO()

StudentLgotnik commented on src/main/java/student/enterprise/project/SEPApplication.java:
```diff
@@ -1,13 +1,23 @@
-package student.enterprise.project;
+package student.enterprise;
```
Что мейн делает в другом пакете?

StudentLgotnik commented on src/main/java/student/enterprise/project/dto/GroupDefaultSchedule.java:
```diff
@@ -0,0 +1,63 @@
+
+
+@AllArgsConstructor
+public class GroupDefaultSchedule {
```
1. Если класс является ДТОшкой, то добавь ДТО в название

2. Для каких целей этот класс?

alexzhyshko replied:
2. Класс для хранения расписания на две недели и изменения пар.

spanasiuk replied:
у нас же дефолтная группа = обыкновенная группа, у которой просто заданы repeatable ченджи

StudentLgotnik commented on src/main/java/student/enterprise/project/dto/LecturerDTO.java:
```diff
@@ -0,0 +1,136 @@
+		this.degree = ld;
+	}
+
+	public Long getId() {
```
Уже есть аннотация Data

StudentLgotnik commented on src/main/java/student/enterprise/project/dto/LessonDTO.java:
```diff
@@ -0,0 +1,52 @@
+
+    private Integer number;
+    
+    public LocalTime getStartTime() {
```
Снова Data

StudentLgotnik commented on src/main/java/student/enterprise/project/dto/RoomDTO.java:
```diff
@@ -0,0 +1,40 @@
+	    
+	int building;
+
+	public Long getId() {
```
Data

StudentLgotnik commented on src/main/java/student/enterprise/project/dto/SubjectDTO.java:
```diff
@@ -0,0 +1,45 @@
+
+    private String name;
+
+	public Long getId() {
```
Data

StudentLgotnik commented on src/main/java/student/enterprise/project/service/RozkladController.java:
```diff
@@ -0,0 +1,190 @@
+			e.printStackTrace();
+		}
+		
+		String weeks = dataJSON.get("weeks").toString();
```
Зачем конвертить в стрингу а потом снова парсить в джейсон, у JSONObject есть метод getJSONObject()

alexzhyshko commented:
ChangeDTO is used to store all information about lesson. To manipulate different lessons (add and delete) we should not segregate ChangeDTO and static lesson dto, it should be one object.
Used Apache Http Client and JSON.simple library.
Developed Http Request class which stores methods needed for making requests. 
Developed Controller class which implements KPIrozklad interface and is used to get new DTO called GroupDefaultSchedule(it will store the whole, two-week, schedule). Http Request class makes request and returns JSON response, controller parses JSON into ChangeDTO and stores them into DefaultGroupSchedule.
Also, added RoomDTO for storing room information and added getters and setters for every DTO class to easily manage all DTOs using new methods in ChangeDTO.

StudentLgotnik commented on src/main/java/student/enterprise/project/SEPApplication.java:
```diff
@@ -1,13 +1,23 @@
-		SpringApplication.run(SEPApplication.class, args);
+		SpringApplication.run(SepApplication.class, args);
+		
+		RozkladController rc = new RozkladController();
```
Зачем это тут?
Если ты хотел проверить работоспосопособность своего метода, для этого тесты пишутся

spanasiuk commented on src/main/java/student/enterprise/project/SEPApplication.java:
```diff
@@ -1,13 +1,23 @@
+
 @SpringBootApplication
-public class SEPApplication {
+public class SepApplication {
```
переименовывай через рефакторинг

spanasiuk commented on src/main/java/student/enterprise/project/dto/GroupDefaultSchedule.java:
```diff
@@ -0,0 +1,63 @@
+
+@AllArgsConstructor
+public class GroupDefaultSchedule {
+	private List<List<ChangeDTO>> days = new ArrayList<>();
```
Категорически неправильный подход.
Как ориентироваться по такому списку?
Надо использовать мапу хотя бы

spanasiuk commented on src/main/java/student/enterprise/project/dto/LecturerDTO.java:
```diff
@@ -0,0 +1,136 @@
+
+		LecturerDegree ld = null;
+		switch (degree) {
+		case "��������":
+			ld = LecturerDegree.PROFESSOR;
+			break;
+		case "������":
+			ld = LecturerDegree.DOCENT;
+			break;
+		case "��������":
+			ld = LecturerDegree.ASSISTANT;
+			break;
+		case "������� ��������":
+			ld = LecturerDegree.ELDERTEACHER;
+			break;
+		case "��������":
```
мне кажется, у тебя слетела кодировка

spanasiuk commented on src/main/java/student/enterprise/project/dto/LecturerDTO.java:
```diff
@@ -0,0 +1,136 @@
+	public void setDegree(String degree) {
+		LecturerDegree ld = null;
+		switch (degree) {
+		case "��������":
+			ld = LecturerDegree.PROFESSOR;
+			break;
+		case "������":
+			ld = LecturerDegree.DOCENT;
+			break;
+		case "��������":
+			ld = LecturerDegree.ASSISTANT;
+			break;
+		case "������� ��������":
+			ld = LecturerDegree.ELDERTEACHER;
+			break;
+		case "��������":
```
кодировка

spanasiuk commented on src/main/java/student/enterprise/project/service/HttpRequestSender.java:
```diff
@@ -0,0 +1,103 @@
+import java.util.Arrays;
+import java.util.List;
+
+public class HttpRequestSender {
```
Зачем это?
Ты вручную имплементил  GET запрос?
RestTemplate же куда удобнее - просто создал экземпляр и сделал .getForObject()

spanasiuk commented on src/main/java/student/enterprise/project/service/RozkladController.java:
```diff
@@ -0,0 +1,190 @@
+	
+	private DayOfWeek dayConverter(String dayUkrName) {
+		switch(dayUkrName) {
+		case "��������":
+			return DayOfWeek.MONDAY;
+		case "³������":
+			return DayOfWeek.TUESDAY;
+		case "������":
+			return DayOfWeek.WEDNESDAY;
+		case "������":
+			return DayOfWeek.THURSDAY;
+		case "ϒ������":
+			return DayOfWeek.FRIDAY;
+		case "������":
+			return DayOfWeek.SATURDAY;
```
кодировка.
И это лучше имплементить на уровне энама, а затем DayOfWeek.getByName(dayUkrName)

alexzhyshko replied:
Енам DayOfWeek уже есть в джаве, его пересоздавать с новым методом глупо.
Подумаю как иначе внедрить слова на украинском языке, видимо браузер не воспринимает кириллицу. У себя в pom.xml ставил cp1251.
